src/main/scala/clocks/Bundles.scala

```diff
@@ -0,0 +1,16 @@
+// See LICENSE.SiFive for license details.
+package freechips.rocketchip.clocks
+
+import chisel3._
+import freechips.rocketchip.util.HeterogeneousBag
+
+class ClockBundle(val params: ClockBundleParameters) extends Bundle
+{
+  val clock = Clock()
+  val reset = Bool()
```

Comment:
could this be of type `Reset`? instead of hard-coding as `Bool`?

Reply:
Good idea. I think it could be. What I need to work out is, is it just of type `Reset()`, or should it actually be parameterized specifically to be either `AsyncReset` or `Bool`? Or maybe we actually want it to be tri-valued like `knownToBeSynchronized: Option[Boolean]`.

Also, it seems that synchronous in this context must mean "synchronized with respect to this particular clock", but is that more specific than the general chisel3 concept of `AsyncReset`? Or is that always defined relative to a particular clock too? @jackkoenig any opinions?

Reply:
Talking with @jackkoenig IRL our current feeling is that actually this should be explicitly parameterized as a handshake between source and sinks, which then makes each bundle emitted concretely contain either Bool or AsyncReset. Argument by analogy here is avoidance of width inference across diplomatically emitted `Record` subtypes.

Reply:
And to answer my above question, reset is always assumed to be synchronously deasserted with respect to the clock of the "module" in question.